pkg/api/controller.go

```diff
@@ -175,7 +179,9 @@ func (c *Controller) Login(w http.ResponseWriter, r *http.Request, body LoginJSO
 	secret := c.Auth.SecretStore().SharedSecret()
 	// user.Username will be different from username/access_key_id on
 	// LDAP login.  Use the stored value.
-	tokenString, err := GenerateJWT(secret, user.ID, loginTime, expires)
+	// Audience and email set here to be an empty string for backward compatibility
+	tokenString, err :=
+		GenerateJWT(secret, LoginAudience, user.ID, *user.Email, loginTime, expires)
```

Comment:
1. We didn't had to change how we generate tokens today
2. if the email is optional - you need to use the email value if it is not nil

Reply:
Fixed indeed, as you suggested to split into separate functions solved these problems as well.